### PR TITLE
[RMP-367] change dropdown opening direction based on the element posi…

### PR DESCRIPTION
**Pull request by mojith**
…tion

[RMP-367] fix dropdown position when there is no room in the bottom

### CODE REVIEW

File: components/src/directives/dropdown-direction/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {Directive, DirectiveBinding} from 'vue';
+
+function fixPosition(el: HTMLElement) {
+  el.classList.remove('--positon-top');
+  el.classList.remove('--positon-bottom');
```
Comment (Super-Chama) [inline components/src/directives/dropdown-direction/index.ts:5]: can be refactored to `el.classList.remove('--positon-top', '--positon-bottom');`

File: components/src/directives/dropdown-direction/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+}
+
+function handleScroll(this: any) {
+  fixPosition(this);
+}
+
+const dropdownDirectionDirective: Directive = {
+  mounted(el: HTMLElement) {
+    window.addEventListener('scroll', handleScroll.bind(el));
```
Comment (Super-Chama) [inline components/src/directives/dropdown-direction/index.ts:23]: can be refactored to
`window.addEventListener('scroll', fixPosition(el));`

and can remove `handleScroll`

File: components/src/directives/dropdown-direction/index.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    fixPosition(el);
+  },
+  updated(el: HTMLElement) {
```
[the remark this answers is not in the document]
Comment (mojith) [reply components/src/directives/dropdown-direction/index.ts:25]: updated is needed for autocompletes.  when typing it shows only one element in the dropdown ('...searching' string) and then it shows the matching results, This kind of DOM changes need to be captured.

File: components/src/directives/dropdown-direction/index.ts (hunk 1 above)
Comment (mojith) [reply components/src/directives/dropdown-direction/index.ts]: done

File: components/src/directives/dropdown-direction/index.ts (hunk 2 above)
Comment (mojith) [reply components/src/directives/dropdown-direction/index.ts:22]: type of the arguments which are expected in 'handleScroll' listener handler function and 'fixPosition' are different.
When registering an event, we can't directly pass values to the listener handler function.

**Review by Super-Chama — APPROVED** [review]

File: components/src/directives/dropdown-direction/index.ts (hunk 1 above)
Comment (Super-Chama) [reply components/src/directives/dropdown-direction/index.ts]: Please mark resolved

**Pull request merged**